iocore/hostdb/HostDB.cc

```diff
@@ -1592,7 +1592,10 @@ HostDBContinuation::probeEvent(int /* event ATS_UNUSED */, Event *e)
   EThread *t = e ? e->ethread : this_ethread();
 
   MUTEX_TRY_LOCK(lock, action.mutex, t);
-  if (!lock.is_locked()) {
+  // Go ahead and grab the continuation mutex or just grab the action mutex again of there is no continuation mutex
+  MUTEX_TRY_LOCK(lock2, (action.continuation && action.continuation->mutex) ? action.continuation->mutex : action.mutex, t);
```

Comment:
Does this only happen with the global session pool?  I was talking to @zwoop about this earlier today.

Reply:
I have only tested with the global pool environment.